gnomad_constraint/utils/constraint.py

```diff
@@ -734,6 +734,52 @@ def calculate_mu_by_downsampling(
     return annotate_mutation_type(ht)
 
 
+def add_oe_lof_upper_rank_and_bin(
+    ht: hl.Table, use_mane_select_instead_of_canonical: bool = False
+) -> hl.Table:
+    """
+    Compute the rank, decile, and sextile of the lof oe upper confidence interval for MANE Select or canonical ensembl transcripts.
+
+    :param ht: Input Table with the value for the lof oe upper confidence interval stored in ht.lof.oe_ci.upper.
+    :param use_mane_select_instead_of_canonical: Use MANE Select rather than canonical transcripts for filtering the Table.
+    :return: Table with anntotations added for 'upper_rank', 'upper_bin_decile', and 'upper_bin_sextile'.
+    """
+    if use_mane_select_instead_of_canonical:
+        transcript_filter = ht.mane_select
```

Comment:
I believe some transcripts don't have a MANE select version, and for those, we'd want to calculate metrics on the canonical annotation. maybe this section could be something like
```
if use_mane_select_instead_of_canonical:
    ht = ht.annotate(transcript_filter=hl.if_else(hl.is_defined(ht.mane_select), ht.mane_select, ht.canonical)
    transcript_filter_expr = ht.transcript_filter
```

Reply:
was that distinction made at the transcript level or gene level? I thought there were some metrics where if the gene didn't have any mane select transcripts, then canonical transcripts were used?

Reply:
ah yeah I think this was made at the gene level. maybe I was misunderstanding, so just to double check: should this function get called twice to cover both cases? (once for MANE select where available and another time to run on canonical?)